webui/react/src/services/apiBuilder.ts

```diff
@@ -54,3 +59,22 @@ export function generateApi<Input, Output>(api: Api<Input, Output>) {
     }
   };
 }
+
+export const consumeStream = async <T = unknown>(
+  fetchArgs: DetSwagger.FetchArgs,
+  onEvent: (event: T) => void,
+  onFinish: () => void): Promise<void> => {
+  let response;
+  try {
+    response = await fetch(serverAddress() + fetchArgs.url, fetchArgs.options);
+  } catch (e) {
+    return processApiError(fetchArgs.url, e);
```

Comment:
thought: My initial thought is that we call `onFinish` here if an error occurs so we can have any clean up code called. However, if we have an `onError` option, then that might suffice.

Reply:
we could cal onFinish onClose in that case ?

Reply:
~actually that will give the user less control compared to not calling that in case of error. which lets the user handle it ~

scratch that. even the on finish is unnecessary. as this is a promise that will "finish" when it's resolved.

Reply:
nit: the `return` should probably just be dropped from `return processApiError(fetchArgs.url, e);` since it's a void function. There is nothing that gets returned.